events/handlers.js

```diff
@@ -0,0 +1,184 @@
+const path = require("path");
+const createCsvWriter = require("csv-writer").createObjectCsvWriter;
+const csv = require("csv-parser");
+const fs = require("fs");
+const converter = require("json-2-csv");
+const { EOF } = require("dns");
+const lockFile = require("lockfile");
+
+const filePath = path.join(__dirname, "events.csv");
+const headerFile = [
+  { id: "id", title: "id" },
+  { id: "title", title: "title" },
+  { id: "location", title: "location" },
+  { id: "timestamp", title: "timestamp" },
+];
+
+function getEventById(req, res) {
+  lockEvents(req, res, (req, res) => {
+    let event = {};
+    fs.createReadStream(filePath)
+      .pipe(csv())
+      .on("data", (data) => {
+        // if we found one row no need to check others
+        if (Object.keys(event).length !== 0) {
+          return;
+        }
+        if (data.id == req.params.eventId) {
+          event = data;
+        }
+      })
+      .on("end", () => {
+        if (Object.keys(event).length === 0) {
+          res.status(404).send("event not found");
+        } else {
+          res.send(event);
+        }
+      });
+  });
+}
+
+function getEvents(req, res) {
+  lockEvents(req, res, (req, res) => {
+    let results = [];
+    fs.createReadStream(filePath)
+      .pipe(csv())
+      .on("data", (data) => {
+        if (
+          req.query.location === undefined ||
+          req.query.location == data.location
+        ) {
+          results.push(data);
+        }
+      })
+      .on("end", () => {
+        res.send(results);
+      });
+  });
+}
+
+function createEvent(req, res) {
+  lockEvents(req, res, (req, res) => {
+    // just add at the end of the file
+    fs.open(filePath, "a", 666, function (e, id) {
+      // convert JSON array to CSV string
+      converter.json2csv(
+        req.body,
+        (err, csv) => {
+          if (err) {
+            res.status(500).send();
+            return;
+          }
+          const row = "\n" + csv;
+          fs.write(id, row, null, "utf8", function () {
+            fs.close(id, function () {
```

Comment:
yeah, since Promises is new for me, I don't have experience with how to organize the code in the appropriate style. I will make refactoring a little bit later